php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1539 @@
+        f.defaultClause(
+          b.token(PHPKeyword.DEFAULT),
+          b.firstOf(b.token(PHPPunctuator.COLON), b.token(PHPPunctuator.SEMICOLON)),
+          b.zeroOrMore(INNER_STATEMENT())
+        )
+      ));
+  }
+
+  public WhileStatementTree WHILE_STATEMENT() {
+    return b.<WhileStatementTree>nonterminal(PHPLexicalGrammar.WHILE_STATEMENT)
+      .is(b.firstOf(
+        f.whileStatement(
+          b.token(PHPKeyword.WHILE),
+          PARENTHESIZED_EXPRESSION(),
+          STATEMENT()
+        ),
+        f.alternativeWhileStatement(
+          b.token(PHPKeyword.WHILE),
+          PARENTHESIZED_EXPRESSION(),
+          b.token(PHPPunctuator.COLON),
+          b.zeroOrMore(INNER_STATEMENT()),
+          b.token(PHPKeyword.ENDWHILE),
+          EOS()
+        )
+      ));
+  }
+
+  public DoWhileStatementTree DO_WHILE_STATEMENT() {
+    return b.<DoWhileStatementTree>nonterminal(PHPLexicalGrammar.DO_WHILE_STATEMENT)
+      .is(f.doWhileStatement(
+        b.token(PHPKeyword.DO),
+        STATEMENT(),
+        b.token(PHPKeyword.WHILE),
+        PARENTHESIZED_EXPRESSION(),
+        EOS()
+      ));
+  }
+
+  public IfStatementTree IF_STATEMENT() {
+    return b.<IfStatementTree>nonterminal(PHPLexicalGrammar.IF_STATEMENT)
+      .is(b.firstOf(STANDARD_IF_STATEMENT(), ALTERNATIVE_IF_STATEMENT()));
+  }
+
+  public IfStatementTree STANDARD_IF_STATEMENT() {
+    return b.<IfStatementTree>nonterminal(PHPLexicalGrammar.STANDARD_IF_STATEMENT)
+      .is(f.ifStatement(
+        b.token(PHPKeyword.IF),
+        PARENTHESIZED_EXPRESSION(),
+        STATEMENT(),
+        b.zeroOrMore(ELSEIF_CLAUSE()),
+        b.optional(ELSE_CLAUSE())
+      ));
+  }
+
+  public IfStatementTree ALTERNATIVE_IF_STATEMENT() {
+    return b.<IfStatementTree>nonterminal(PHPLexicalGrammar.ALTERNATIVE_IF_STATEMENT)
+      .is(f.alternativeIfStatement(
+        b.token(PHPKeyword.IF),
+        PARENTHESIZED_EXPRESSION(),
+        b.token(PHPPunctuator.COLON),
+        b.zeroOrMore(INNER_STATEMENT()),
+        b.zeroOrMore(ALTERNATIVE_ELSEIF_CLAUSE()),
+        b.optional(ALTERNATIVE_ELSE_CLAUSE()),
+        b.token(PHPKeyword.ENDIF),
+        EOS()
+      ));
+  }
+
+  public ElseClauseTree ELSE_CLAUSE() {
+    return b.<ElseClauseTree>nonterminal(PHPLexicalGrammar.ELSE_CLAUSE)
+      .is(f.elseClause(b.token(PHPKeyword.ELSE), STATEMENT()));
+  }
+
+  public ElseifClauseTree ELSEIF_CLAUSE() {
+    return b.<ElseifClauseTree>nonterminal(PHPLexicalGrammar.ELSEIF_CLAUSE)
+      .is(f.elseifClause(
+        b.token(PHPKeyword.ELSEIF),
+        PARENTHESIZED_EXPRESSION(),
+        STATEMENT()
+      ));
+  }
+
+  public ElseClauseTree ALTERNATIVE_ELSE_CLAUSE() {
+    return b.<ElseClauseTree>nonterminal(PHPLexicalGrammar.ALTERNATIVE_ELSE_CLAUSE)
+      .is(f.alternativeElseClause(
+        b.token(PHPKeyword.ELSE),
+        b.token(PHPPunctuator.COLON),
+        b.zeroOrMore(INNER_STATEMENT())
+      ));
+  }
+
+  public ElseifClauseTree ALTERNATIVE_ELSEIF_CLAUSE() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)